octobot/community/errors_upload/error_model.py

```diff
@@ -0,0 +1,56 @@
+#  Drakkar-Software OctoBot
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import traceback
+
+
+class Error:
+    """
+    Error is the error data wrapper use to upload errors to the error server
+    """
+
+    def __init__(self, error: Exception, title: str, timestamp: float, metrics_id: str):
+        self.error: Exception = error
+        self.title: str = title
+        self.timestamps: list = [timestamp]
+        self.metrics_id: str = metrics_id
+        self.type: str = self.error.__class__.__name__ if self.error else ""
+        self.stacktrace: list = traceback.format_exception(
+            etype=type(self.error), value=self.error, tb=self.error.__traceback__
+        )[1:] if self.error else []
+
+    def to_dict(self) -> dict:
+        """
+        Return the dict serialization of self
+        """
+        return {
+            "title": self.title,
+            "type": self.type,
+            "stacktrace": self.stacktrace,
+            "timestamps": self.timestamps,
+            "metricsid": self.metrics_id,
+        }
+
+    def is_equivalent(self, other) -> bool:
```

Comment:
I think we should implement `def __eq__(self, object)` instead

Reply:
I also considered it and since it's not a real equals, I concluded that it's better not to do it. However we don't use `==` anywhere else yet so it might be better for now and revert it if we need to later ?

Reply:
Ok, I was just wondering if you had thought about it :wink: